Library/Homebrew/test/cask/audit_spec.rb

```diff
@@ -582,87 +588,87 @@ def tmp_cask(name, text)
       end
 
       context "when the download uses GitHub releases and does not have an appcast" do
         let(:cask_token) { "github-without-appcast" }
 
-        it { is_expected.to warn_with(appcast_warning) }
+        it { is_expected.to fail_with(message) }
       end
 
       context "when the download is hosted on SourceForge and has an appcast" do
         let(:cask_token) { "sourceforge-with-appcast" }
 
-        it { is_expected.not_to warn_with(appcast_warning) }
+        it { is_expected.not_to fail_with(message) }
       end
 
       context "when the download is hosted on SourceForge and does not have an appcast" do
         let(:cask_token) { "sourceforge-correct-url-format" }
 
-        it { is_expected.to warn_with(appcast_warning) }
+        it { is_expected.to fail_with(message) }
       end
 
       context "when the download is hosted on DevMate and has an appcast" do
         let(:cask_token) { "devmate-with-appcast" }
 
-        it { is_expected.not_to warn_with(appcast_warning) }
+        it { is_expected.not_to fail_with(message) }
       end
 
       context "when the download is hosted on DevMate and does not have an appcast" do
         let(:cask_token) { "devmate-without-appcast" }
 
-        it { is_expected.to warn_with(appcast_warning) }
+        it { is_expected.to fail_with(message) }
       end
 
       context "when the download is hosted on HockeyApp and has an appcast" do
         let(:cask_token) { "hockeyapp-with-appcast" }
 
-        it { is_expected.not_to warn_with(appcast_warning) }
+        it { is_expected.not_to fail_with(message) }
       end
 
       context "when the download is hosted on HockeyApp and does not have an appcast" do
         let(:cask_token) { "hockeyapp-without-appcast" }
 
-        it { is_expected.to warn_with(appcast_warning) }
+        it { is_expected.to fail_with(message) }
       end
     end
 
     describe "latest with appcast checks" do
-      let(:warning_msg) { "Casks with an appcast should not use version :latest" }
+      let(:message) { "Casks with an appcast should not use version :latest" }
 
       context "when the Cask is :latest and does not have an appcast" do
         let(:cask_token) { "version-latest" }
 
-        it { is_expected.not_to warn_with(warning_msg) }
+        it { is_expected.not_to fail_with(message) }
       end
 
       context "when the Cask is versioned and has an appcast" do
         let(:cask_token) { "with-appcast" }
 
-        it { is_expected.not_to warn_with(warning_msg) }
+        it { is_expected.not_to fail_with(message) }
       end
 
       context "when the Cask is :latest and has an appcast" do
         let(:cask_token) { "latest-with-appcast" }
 
-        it { is_expected.to warn_with(warning_msg) }
+        it { is_expected.to fail_with(message) }
       end
     end
 
     describe "denylist checks" do
-      context "when the Cask isn't disallowed" do
+      context "when the Cask is allowed" do
```

Comment:
```suggestion
      context "when the Cask is not on the denylist" do
```
A cask can pass the denylist check but still not be allowed for other reasons.